includes/Core/Util/URL.php

```diff
@@ -147,13 +149,13 @@ public static function permute_site_hosts( $hostname ) {
 			// An ASCII hostname can only be non-IDN or punycode-encoded.
 			if ( $is_ascii ) {
 				// If the hostname is in punycode encoding, add the decoded version to the list of hosts.
-				if ( 0 === strpos( $hostname, Punycode::PREFIX ) || false !== strpos( $hostname, '.' . Punycode::PREFIX ) ) {
-					$host_decoded = $punycode->decode( $hostname );
+				if ( 0 === strpos( $hostname, self::PUNYCODE_PREFIX ) || false !== strpos( $hostname, '.' . self::PUNYCODE_PREFIX ) ) {
+					$host_decoded = idn_to_utf8( $hostname, IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46 );
 					array_push( $hosts, $host_decoded, "www.$host_decoded" );
 				}
 			} else {
 				// If it's not ASCII, then add the punycode encoded version.
-				$host_encoded = $punycode->encode( $hostname );
+				$host_encoded = idn_to_ascii( $hostname, IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46 );
```

Comment:
The same comment applies here, we should stick with the defaults for `$flags` and `$variant`.

```suggestion
				$host_encoded = idn_to_ascii( $hostname );
```

Reply:
Removing these fails `composer lint` with the comment:

> The default value of the idn_to_ascii() $variant parameter has changed from INTL_IDNA_VARIANT_2003 to INTL_IDNA_VARIANT_UTS46 in PHP 7.4. For optimal cross-version compatibility, the $variant parameter should be explicitly set.

I've pushed adding in a `phpcs:ignore` comment so that this function will always use the defaults of the php version it's loaded with.

Reply:
Top stuff, cheers.